.taskmaster/docs/MASTER-blog-post.md

```diff
@@ -0,0 +1,1767 @@
+    internal let apiToken: String
+    // All let properties - inherently thread-safe
+}
+```
+
+**3. Actors for Thread Safety**:
+```swift
+actor ServerToServerAuthManager: TokenManager {
+    private var privateKey: P256.Signing.PrivateKey
+
+    func signRequest(_ request: URLRequest) async throws -> URLRequest {
+        // Thread-safe by design
+    }
+}
+```
+
+**4. Protocol-Oriented Design**:
+- TokenManager protocol
+- Three implementations: APITokenManager, WebAuthTokenManager, ServerToServerAuthManager
+- Easy testing with mocks
+- Flexible implementation swapping
+
+**5. Typed Throws (Swift 6)**:
+```swift
+func validateCredentials() async throws(TokenManagerError) -> Bool
+```
+
+### Section 4.4: CustomFieldValue - The Design Decision (~150 words)
+
+**The Question**: Override generated `FieldValue` or use it as-is?
+
+**The CloudKit Quirk**:
+- ASSET type: Full metadata (size, checksum, download URL)
+- ASSETID type: Just a string reference
+- Generated code can't distinguish these automatically
+
+**The Decision**: Override with `CustomFieldValue`
+
+**Configuration**:
+```yaml
+typeOverrides:
+  schemas:
+    FieldValue: CustomFieldValue
+```
+
+**CustomFieldValue Implementation**:
+```swift
+internal struct CustomFieldValue: Codable, Hashable, Sendable {
+    internal enum FieldTypePayload: String, Codable {
+        case string = "STRING"
+        case int64 = "INT64"
+        case asset = "ASSET"
+        case assetid = "ASSETID"  // CloudKit quirk
+        // ... more types
+    }
+
+    internal let value: CustomFieldValuePayload
+    internal let type: FieldTypePayload?
+}
+```
+
+**Claude's Role**:
+- Suggested type override approach
+- Generated test cases for all field value types
+- Found edge cases (empty lists, nil values, malformed data)
+
+### Section 4.5: Security Built-In (~100 words)
+
+**Automatic Credential Masking**:
+
+```swift
+internal enum SecureLogging {
+    internal static func maskToken(_ token: String) -> String {
+        guard token.count > 8 else { return "***" }
+        let prefix = token.prefix(4)
+        let suffix = token.suffix(4)
+        return "\(prefix)***\(suffix)"
+    }
+}
+```
+
+**In LoggingMiddleware**:
+```swift
+private func formatQueryValue(for item: URLQueryItem) -> String {
+    guard let value = item.value else { return "nil" }
+
+    if item.name.lowercased().contains("token") ||
+       item.name.lowercased().contains("key") {
+        return SecureLogging.maskToken(value)
+    }
+
+    return value
+}
+```
+
+**Output**:
+```
+🌐 CloudKit Request: POST /database/1/iCloud.com.example/production/private/records/query
+  ckAPIToken: c34a***7d9f
+  ckWebAuthToken: 9f2e***4b1a
+```
+
+**Claude Generated Logging**: I added security constraints (never log full tokens, mask private keys)
+
+> **Lessons Learned**: For insights about working with Claude on this architecture, see Part 6: Lessons Learned.
+
+---
+
+## PART 5: The Three-Month Journey with Claude Code (800 words)
```

Comment:
I'm not sure what this section is for. Is there a way we can remove it for keep the journey narrative interspersed?